airbyte-integrations/connectors/source-iterable/source_iterable/source.py

```diff
@@ -53,101 +42,68 @@
+source connector.
 
-class SourceIterable(AbstractSource):
-    """
-    Note: there are some redundant endpoints
-    (e.g. [`export/userEvents`](https://api.iterable.com/api/docs#export_exportUserEvents)
-    and [`events/{email}`](https://api.iterable.com/api/docs#events_User_events)).
-    In this case it's better to use the one which takes params as a query param rather than as part of the url param.
-    """
+WARNING: Do not modify this file.
+"""
 
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
-        try:
-            authenticator = TokenAuthenticator(token=config["api_key"], auth_header="Api-Key", auth_method="")
-            list_gen = Lists(authenticator=authenticator).read_records(sync_mode=SyncMode.full_refresh)
-            next(list_gen)
-            return True, None
-        except Exception as e:
-            return False, f"Unable to connect to Iterable API with the provided credentials - {e}"
+# Declarative Source
+class SourceIterable(YamlDeclarativeSource):
+    def __init__(self):
+        super().__init__(**{"path_to_yaml": "manifest.yaml"})
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
-        def all_streams_accessible():
-            access_check_stream = AccessCheck(authenticator=authenticator)
-            try:
-                next(read_full_refresh(access_check_stream), None)
-            except requests.exceptions.RequestException as e:
-                if e.response.status_code == requests.codes.UNAUTHORIZED:
-                    return False
-                raise
-            return True
+        streams = super().streams(config=config)
 
         authenticator = TokenAuthenticator(token=config["api_key"], auth_header="Api-Key", auth_method="")
         # end date is provided for integration tests only
         start_date, end_date = config["start_date"], config.get("end_date")
         date_range = {"start_date": start_date, "end_date": end_date}
-        streams = [
-            Campaigns(authenticator=authenticator),
-            CampaignsMetrics(authenticator=authenticator, **date_range),
-            Channels(authenticator=authenticator),
-            Lists(authenticator=authenticator),
-            MessageTypes(authenticator=authenticator),
-            Metadata(authenticator=authenticator),
-            Templates(authenticator=authenticator, **date_range),
-        ]
-        # Iterable supports two types of Server-side api keys:
-        # - read only
-        # - server side
-        # The first one has a limited set of supported APIs, so others are filtered out here.
-        # A simple check is done - a read operation on a stream that can be accessed only via a Server side API key.
-        # If read is successful - other streams should be supported as well.
-        # More on this - https://support.iterable.com/hc/en-us/articles/360043464871-API-Keys-
-        if all_streams_accessible():
```

Comment:
Should we still be using this check? If not, why not?

Reply:
No, default availability strategy will take care about that.